File: maximizeProfit.c
#include<stdio.h>

int maxProfit(int* prices, int pricesSize) {
    int minIndex = prices[0];
    int maxPrice = 0;

    for(int i=0; i<pricesSize; i++){
        if(minIndex<prices[i]){
            minIndex = i;
        }
    }

    printf("%d", minIndex);

    for(int i=minIndex; i<pricesSize; i++){
        if(prices[i]>maxPrice){
            maxPrice = i;
        }
    }

    return maxPrice;
}

int main(){
    int price[] = {7,1,5,3,6,4};
    int size = sizeof(price)/sizeof(price[0]);
    int result = maxProfit(price, size);
    printf("%d", result);
}

File: maximizeProfit.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
  if (prices.length < 2) return 0;

  let minPrice = prices[0];
  let maxProfit = 0;

  for (let i = 1; i < prices.length; i++) {
    const profit = prices[i] - minPrice;

    maxProfit = Math.max(maxProfit, profit);

    minPrice = Math.min(minPrice, prices[i]);
  }

  return maxProfit;
};

const prices = [7, 1, 5, 3, 6, 4];
const result = maxProfit(prices);
console.log(result);
